apps/meteor/app/ui-utils/client/lib/LegacyRoomManager.ts

```diff
@@ -96,6 +98,11 @@ const computation = Tracker.autorun(() => {
 			void RoomHistoryManager.getMoreIfIsEmpty(record.rid);
 
 			if (room) {
+				const subscription = Subscriptions.findOne({ rid: record.rid }, { reactive: false });
+				if (!hasPermission('preview-c-room') && room.t === 'c' && !subscription) {
+					return;
+				}
+
 				if (record.streamActive !== true) {
```

Comment:
Shouldn't `if (record.streamActive !== true) {` check come first?

Reply:
No, because if stream not active we will have exactly what we are trying to avoid, load the stream.
In this case we have another check who comes first, and it's related to preview the public room with no subscriptions...